### PR TITLE
Simplify Configure calls

### PR DESCRIPTION
**Pull request by kayoub5**
Nothing fancy, just trying to reduce the number of ifs

### CODE REVIEW

File: SharpPcap/LibPcap/CaptureFileWriterDevice.cs [hunk 1]
```diff
@@ -111,11 +111,6 @@ public override void Close()
         /// </summary>
         public override void Open(DeviceConfiguration configuration)
         {
-            if (configuration.Snaplen > Pcap.MAX_PACKET_SIZE)
-            {
-                configuration.RaiseConfigurationFailed("snaplen", new InvalidOperationException("Snaplen > Pcap.MAX_PACKET_SIZE"));
-            }
-
```
Comment (kayoub5) [inline SharpPcap/LibPcap/CaptureFileWriterDevice.cs:118]: @chmorgan after checking libpcap docs, there is no statement that says spanlen must not be larger than 65536.
All what the docs says, if you set it to 65536, then it is big enough for any known traffic.

Comment (chmorgan) [reply SharpPcap/LibPcap/CaptureFileWriterDevice.cs:118]: I think that is the case. I'm not sure it always was or not. Might have been from the original library even.

**Comment by kayoub5** [comment]
@chmorgan waiting for your review

**Review by chmorgan — CHANGES_REQUESTED** [review]
Can you rebase on master now that the timestamp changes landed and I'll re-review?

File: SharpPcap/LibPcap/PcapUnmanagedStructures.cs [hunk 2]
```diff
@@ -219,18 +219,6 @@ public struct pcap_pkthdr_macosx
 
         #endregion
 
-        /// <summary>
```
Comment (chmorgan) [inline SharpPcap/LibPcap/PcapUnmanagedStructures.cs:222]: it could be helpful to put this change in its own commit to explain the reasoning here, it seems unrelated to removing the MAX_PACKET checks.

**Review by chmorgan — CHANGES_REQUESTED** [review]

File: SharpPcap/LibPcap/PcapUnmanagedStructures.cs (hunk 2 above)
Comment (kayoub5) [reply SharpPcap/LibPcap/PcapUnmanagedStructures.cs:222]: It is related to MAX_PACKET_SIZE, see the attribute `[MarshalAs(UnmanagedType.ByValArray, SizeConst = SharpPcap.Pcap.MAX_PACKET_SIZE)]`

I should have changed the attribute, but the struct was not used so I just deleted it.

Comment (chmorgan) [reply SharpPcap/LibPcap/PcapUnmanagedStructures.cs:222]: ahh k. Can you put that in a separate commit if its unused? maybe remove it prior to adjusting the max_packet stuff, like the middle commit in the series?

**Comment by chmorgan** [comment]
@kayoub5 can we agree that code coverage shouldn't decrease with new PRs? we should be pushing towards having a higher coverage rate if possible. Would it make sense to land the arp test and cde branches and rebase to see if this one improves? I noticed you touch the capture file writer and that has improved coverage on the cde branch.

File: SharpPcap/LibPcap/PcapUnmanagedStructures.cs (hunk 2 above)
Comment (kayoub5) [reply SharpPcap/LibPcap/PcapUnmanagedStructures.cs:222]: Done

**Comment by kayoub5** [comment]
@chmorgan coverage wise, I can't find a line that previously was covered and now is not, the number of non covered lines did not go up. the number of covered lines went down since I deleted more code than I added.

**Review by chmorgan — APPROVED** [review]
